SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,292 @@
+//===-- functions_1d_select.hpp - Functions for tests on simd rvalue select
+//      function. ---------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd 1d select function.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+// for std::ceil
+#include <cmath>
+// for std::numeric_limits
+#include <limits>
+// for std::iota
+#include <numeric>
+
+namespace esimd_test::api::functional::functions {
+
+namespace details {
+
+constexpr int ceil(int a, int b) { return (a % b) > 0 ? a / b + 1 : a / b; }
```

Comment:
Not really required, but just as suggestion:
```suggestion
constexpr int ceil(int a, int b) { return ((a % b) > 0) ? (a / b + 1) : (a / b); }
```

Reply:
this comment applied [here](https://github.com/intel/llvm-test-suite/pull/875/commits/6e3c354f9cdd7463f3b5749cc615b9ff59eaa9ca#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3R29-R31)